src/mgmt/srv_query.c

```diff
@@ -299,3 +298,165 @@ ds_mgmt_smd_list_pools(Mgmt__SmdPoolResp *resp)
 out:
 	return rc;
 }
+
+int
+ds_mgmt_dev_state_query(uuid_t dev_uuid, Mgmt__DevStateResp *resp)
+{
+	struct smd_dev_info	*dev_info;
+	int			 buflen = 10;
+	int			 rc = 0;
+
+	if (uuid_is_null(dev_uuid))
+		return -1;
```

Comment:
I noticed it returns -1 in some cases and DER errors in some other cases, could you explain it a bit?

Reply:
I will make them all -DER errors